### PR TITLE
XW-1733 | fix sending events to GA

### PR DESCRIPTION
**Pull request by tomasznapieralski**
## Links

* http://wikia-inc.atlassian.net/browse/XW-1733

## Changes

* google analytics script is downloaded over https (google recommends)
* google script is loaded in head instead of after body (google recommends)
* events send before GA initializer are stored and resend after initializer

## Reviewers

@Wikia/x-wing

### CODE REVIEW

File: front/common/baseline/tracker-universal-analytics.js [hunk 1]
```diff
@@ -492,6 +506,6 @@ if (typeof window.M.tracker === 'undefined') {
 		_getDimensionsSynced: getDimensionsSynced,
 		_updateTrackedUrl: updateTrackedUrl,
 		_filterQueryParams: filterQueryParams,
-		_dimensions: dimensions,
+		_dimensions: dimensions
```
Comment (vforge) [inline front/common/baseline/tracker-universal-analytics.js]: There's need to remove that comma?

File: server/app/views/_partials/tracking-ua-auth.hbs [hunk 2]
```diff
@@ -1,8 +1,5 @@
 {{#unless queryParams.noexternals}}
 	<script>
-		// Load UA async and flag $script when ready
```
Comment (vforge) [inline server/app/views/_partials/tracking-ua-auth.hbs]: Remember about this https://github.com/Wikia/mercury/blob/dev/server/app/views/_partials/tracking-ua-auth.hbs#L4 - and check auth tracking too

File: front/common/baseline/tracker-universal-analytics.js [hunk 3]
```diff
@@ -30,10 +30,12 @@ if (typeof window.M.tracker === 'undefined') {
 
 (function (M) {
 	let tracked = [],
+		trackEvents = [],
```
Comment (rogatty) [inline front/common/baseline/tracker-universal-analytics.js]: How about `eventsQueue`? Would be more descriptive.

File: front/common/baseline/tracker-universal-analytics.js [hunk 4]
```diff
@@ -460,6 +466,14 @@ if (typeof window.M.tracker === 'undefined') {
 			initAccount(accountSpecial);
 		}
 
+		isInitialized = true;
+
+		trackEvents.forEach((event) => {
+			track(event.category, event.action, event.label, event.value, event.nonInteractive);
+		});
+
+		trackEvents = [];
```
Comment (rogatty) [inline front/common/baseline/tracker-universal-analytics.js]: I suggest moving this line and the loop above to a separate function named like `flushEventsQueue`, just for the readibility.

**Comment by rogatty** [comment]
Apart from two minor comments LGTM.

**Pull request merged**